fix: transient payload example (#2650)
ory/docs: docs/kratos/emails-sms/01_sending-emails-smtp.mdx

</Tabs>
```

The body is built from a Jsonnet template. See [Payload](#payload) for the `ctx` object the template receives and an example
template.

</TabItem>
</Tabs>

## Payload {#payload}

When you use the HTTP delivery strategy, Ory Identities sends each email as an HTTP request whose body is built from a Jsonnet
template. The template receives a `ctx` object that Ory Identities constructs for every message.

### The `ctx` object

> The shape of `ctx` is defined by the `httpDataModel` struct in Ory Kratos.

Reviewer: ```suggestion
The `ctx` properties are defined by the `httpDataModel` struct in Ory Kratos.
```

| `subject`         | String | The rendered email subject.                                                                                          |
| `body`            | String | The rendered plaintext body.                                                                                         |
| `html_body`       | String | The rendered HTML body, when available.                                                                              |
| `template_type`   | String | The template type. See [Events that trigger an email](./05_custom-email-templates.mdx#events-that-trigger-an-email). |
| `template_data`   | Object | The per-template variables. See [Available variables](./05_custom-email-templates.mdx#available-variables).          |
| `message_type`    | String | The message channel, for example `email`.                                                                            |
| `request_headers` | Object | The HTTP headers from the flow request that triggered the email.                                                     |

### Example Jsonnet template

> The following template is the example mailer configuration shipped with Ory Kratos. It maps the most common fields out of `ctx`.

Reviewer: ```suggestion
The following is the mailer configuration template shipped with Ory Kratos. It maps the most common fields from `ctx`.
```

> <CodeFromRemote

Reviewer: The code snippet renders without a readable format

Author: @unatasha8 can you clarify what you mean by this?

this is what it looks like for me

<img width="1053" height="619" alt="Screenshot 2026-07-03 at 09 43 26" src="https://github.com/user-attachments/assets/9e2d60a7-2f91-4092-bc41-942072ea8738" />

Reviewer: That's what I see too! Is that considered readable?

Author: that is what the complete jsonnet function looks like when you use it in your code. 

of course in most cases you wont need the complete ctx object, only a couple values, but I think this is helpful if you write your own and are looking for an example.

also it gives a complete overview of what is available in the mailer ctx.

  lang="jsonnet"
  link="https://github.com/ory/kratos/blob/master/courier/stub/request.config.mailer.jsonnet"
  src="https://raw.githubusercontent.com/ory/kratos/master/courier/stub/request.config.mailer.jsonnet"
/>

> Read fields that the template you target needs from `ctx.template_data`, and read top-level fields such as `recipient` and

Reviewer: ```suggestion
When using your own template, map top-level fields such as `recipient` and `template_type` directly from `ctx`. Map the other fields your template needs from `ctx.template_data`.
```

> `template_type` directly from `ctx`.

Reviewer: ```suggestion

```
